Missing zap-cli fails arm64 builds

The arm64 build of v1.1.0.1 fails with the following error:
```
:: [33/555] ACTION //third_party/connectedhomeip/examples/lighting-app/lighting-common:lighting-common_zapgen_zap_pregen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: FAILED: gen/third_party/connectedhomeip/examples/lighting-app/lighting-common/zap_pregen/zap-generated/access.h gen/third_party/connectedhomeip/examples/lighting-app/lighting-common/zap_pregen/zap-generated/gen_config.h gen/third_party/connectedhomeip/examples/lighting-app/lighting-common/zap_pregen/zap-generated/endpoint_config.h gen/third_party/connectedhomeip/examples/lighting-app/lighting-common/zap_pregen/zap-generated/CHIPClientCallbacks.h gen/third_party/connectedhomeip/examples/lighting-app/lighting-common/zap_pregen/zap-generated/CHIPClusters.h gen/third_party/connectedhomeip/examples/lighting-app/lighting-common/zap_pregen/zap-generated/IMClusterCommandHandler.cpp
:: python ../../third_party/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root ../../ --current-path ../../third_party/connectedhomeip/examples/lighting-app/lighting-common --default-toolchain=//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc --current-toolchain=//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc --capture-output --python-dep-list-files gen/third_party/connectedhomeip/examples/lighting-app/lighting-common/lighting-common_zapgen_zap_pregen_metadata_path_list.txt -- ../../third_party/connectedhomeip/scripts/tools/zap/generate.py --no-prettify-output --templates /build/snapcraft-matter-pi-gpio-commander-28bfd9faac2be3cc31a04dd4da83a95d/parts/connectedhomeip/src/examples/lighting-app/linux/third_party/connectedhomeip/src/app/zap-templates/app-templates.json --output-dir /build/snapcraft-matter-pi-gpio-commander-28bfd9faac2be3cc31a04dd4da83a95d/parts/connectedhomeip/src/examples/lighting-app/linux/out/build/gen/third_party/connectedhomeip/examples/lighting-app/lighting-common/zap_pregen/zap-generated --lock-file /build/snapcraft-matter-pi-gpio-commander-28bfd9faac2be3cc31a04dd4da83a95d/parts/connectedhomeip/src/examples/lighting-app/linux/out/build/zap_gen.lock --parallel /build/snapcraft-matter-pi-gpio-commander-28bfd9faac2be3cc31a04dd4da83a95d/parts/connectedhomeip/src/examples/lighting-app/linux/third_party/connectedhomeip/examples/lighting-app/lighting-common/lighting-app.zap
:: Searching for zcl file from /build/snapcraft-matter-pi-gpio-commander-28bfd9faac2be3cc31a04dd4da83a95d/parts/connectedhomeip/src/examples/lighting-app/linux/third_party/connectedhomeip/examples/lighting-app/lighting-common/lighting-app.zap
:: FAILED TO EXECUTE ZAP GENERATION: No such file or directory - "zap-cli"
:: ********************************************************************************
:: * You may need to install zap. Please ensure one of these applies:
:: * - `zap-cli` is in $PATH. Install from https://github.com/project-chip/zap/releases
:: *   see docs/guides/BUILDING.md for details
:: * - `zap-cli` is in $ZAP_INSTALL_PATH. Use this option if you
:: *   installed zap but do not want to update $PATH
:: * - Point $ZAP_DEVELOPMENT_PATH to your local copy of zap that you
:: *   develop on (to use a developer build of zap)
:: ********************************************************************************
:: [34/555] ACTION //third_party/connectedhomeip/src/platform/Linux/dbus/bluez:bluez_gen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: [35/555] ACTION //third_party/connectedhomeip/src/platform/Linux/dbus/openthread:openthread_gen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: [36/555] ACTION //third_party/connectedhomeip/src/platform/Linux/dbus/wpa:wpa_gen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: [37/555] ACTION //third_party/connectedhomeip/src/platform/Linux/dbus/wpa:wpa_gen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: [38/555] ACTION //third_party/connectedhomeip/examples/lighting-app/lighting-common:lighting-common_codegen_codegen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: ninja: build stopped: subcommand failed.
```
The same issue exists on the chip-tool side, where it utilizes the same version of the Matter SDK: https://github.com/canonical/chip-tool-snap/issues/3

Remote build fails for arm64:
```
:: + [[ arm64 == \a\r\m\6\4 ]]
:: + ZAP_VERSION=v2023.05.04
:: + set -x
:: + mkdir -p /opt/zap-v2023.05.04
:: + git clone https://github.com/project-chip/zap.git /opt/zap-v2023.05.04
:: Cloning into '/opt/zap-v2023.05.04'...
:: + cd /opt/zap-v2023.05.04
:: + git checkout -b v2023.05.04
:: Switched to a new branch 'v2023.05.04'
:: + npm ci
:: npm WARN tarball tarball data for app-builder-bin@3.7.1 (sha512-ql93vEUq6WsstGXD+SBLSIQw6SNnhbDEM0swzgugytMxLp3rT24Ag/jcC80ZHxiPRTdew1niuR7P3/FCrDqIjw==) seems to be corrupted. Trying one more time.
:: npm WARN lifecycle The node binary used for scripts is /snap/bin/node but npm is using /snap/node/6202/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
::
:: > utf-8-validate@5.0.10 install /opt/zap-v2023.05.04/node_modules/utf-8-validate
:: > node-gyp-build
::
:: make: Entering directory '/opt/zap-v2023.05.04/node_modules/utf-8-validate/build'
::   CC(target) Release/obj.target/validation/src/validation.o
::   SOLINK_MODULE(target) Release/obj.target/validation.node
::   COPY Release/validation.node
:: make: Leaving directory '/opt/zap-v2023.05.04/node_modules/utf-8-validate/build'
::
:: > sqlite3@5.1.4 install /opt/zap-v2023.05.04/node_modules/sqlite3
:: > node-pre-gyp install --fallback-to-build
::
:: [sqlite3] Success: "/opt/zap-v2023.05.04/node_modules/sqlite3/lib/binding/napi-v6-linux-glibc-arm64/node_sqlite3.node" is installed via remote
::
:: > core-js@3.27.1 postinstall /opt/zap-v2023.05.04/node_modules/core-js
:: > node -e "try{require('./postinstall')}catch(e){}"
::
[16/Jun/2023:09:36:48 +0000] "CONNECT github.com:443 HTTP/1.1" 200 3772 "-" "-"
:: [96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m
::
:: [96mThe project needs your help! Please consider supporting of core-js:[0m
:: [96m>[94m https://opencollective.com/core-js [0m
:: [96m>[94m https://patreon.com/zloirock [0m
:: [96m>[94m bitcoin: bc1qlea7544qtsmj2rayg0lthvza9fau63ux0fstcz [0m
::
:: [96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m
::
::
:: > canvas@2.11.2 install /opt/zap-v2023.05.04/node_modules/canvas
:: > node-pre-gyp install --fallback-to-build --update-binary
::
:: node-pre-gyp ERR! install response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.11.2/canvas-v2.11.2-node-v72-linux-glibc-arm64.tar.gz
:: node-pre-gyp WARN Pre-built binaries not installable for canvas@2.11.2 and node@12.22.12 (node-v72 ABI, glibc) (falling back to source compile with node-gyp)
:: node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.11.2/canvas-v2.11.2-node-v72-linux-glibc-arm64.tar.gz
:: Package pangocairo was not found in the pkg-config search path.
:: Perhaps you should add the directory containing `pangocairo.pc'
:: to the PKG_CONFIG_PATH environment variable
:: No package 'pangocairo' found
:: gyp: Call to 'pkg-config pangocairo --libs' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
:: gyp ERR! configure error
:: gyp ERR! stack Error: `gyp` failed with exit code: 1
:: gyp ERR! stack     at ChildProcess.onCpExit (/snap/node/6202/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
:: gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
:: gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
:: gyp ERR! System Linux 5.4.0-149-generic
:: gyp ERR! command "/snap/node/6202/bin/node" "/snap/node/6202/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--update-binary" "--module=/opt/zap-v2023.05.04/node_modules/canvas/build/Release/canvas.node" "--module_name=canvas" "--module_path=/opt/zap-v2023.05.04/node_modules/canvas/build/Release" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v72"
:: gyp ERR! cwd /opt/zap-v2023.05.04/node_modules/canvas
:: gyp ERR! node -v v12.22.12
:: gyp ERR! node-gyp -v v5.1.0
:: gyp ERR! not ok
:: node-pre-gyp ERR! build error
:: node-pre-gyp ERR! stack Error: Failed to execute '/snap/node/6202/bin/node /snap/node/6202/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --update-binary --module=/opt/zap-v2023.05.04/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/opt/zap-v2023.05.04/node_modules/canvas/build/Release --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
:: node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/opt/zap-v2023.05.04/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
:: node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
:: node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1022:16)
:: node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
:: node-pre-gyp ERR! System Linux 5.4.0-149-generic
:: node-pre-gyp ERR! command "/snap/node/6202/bin/node" "/opt/zap-v2023.05.04/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--update-binary"
:: node-pre-gyp ERR! cwd /opt/zap-v2023.05.04/node_modules/canvas
:: node-pre-gyp ERR! node -v v12.22.12
:: node-pre-gyp ERR! node-pre-gyp -v v1.0.10
:: node-pre-gyp ERR! not ok
:: Failed to execute '/snap/node/6202/bin/node /snap/node/6202/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --update-binary --module=/opt/zap-v2023.05.04/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/opt/zap-v2023.05.04/node_modules/canvas/build/Release --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
:: npm ERR! code ELIFECYCLE
:: npm ERR! errno 1
:: npm ERR! canvas@2.11.2 install: `node-pre-gyp install --fallback-to-build --update-binary`
:: npm ERR! Exit status 1
:: npm ERR!
:: npm ERR! Failed at the canvas@2.11.2 install script.
:: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
::
:: npm ERR! A complete log of this run can be found in:
:: npm ERR!     /root/.npm/_logs/2023-06-16T09_36_50_920Z-debug.log
'override-build' in part 'lighting' failed with code 1.
```

--- a/README.md
+++ b/README.md
@@ -92,23 +92,14 @@ For the following examples, we use the [Chip Tool snap](https://snapcraft.io/chi
 ### Commissioning
 
 ```bash
-sudo chip-tool pairing ethernet 110 20202021 3840 192.168.1.111 5540
+sudo snap connect chip-tool:avahi-observe
+sudo chip-tool pairing onnetwork 110 20202021
 ```
 
 where:
 
 -   `110` is the assigned node id
 -   `20202021` is the default passcode (pin code) for the lighting app
--   `3840` is the discriminator id
--   `192.168.1.111` is the IP address of the host for the lighting app
--   `5540` the the port for the lighting app
-
-Alternatively, to commission with discovery which works with DNS-SD:
-
-```bash
-sudo snap connect chip-tool:avahi-observe
-sudo chip-tool pairing onnetwork 110 20202021
-```
 
 ### Command
 

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -48,7 +48,7 @@ parts:
     plugin: nil
     source: https://github.com/project-chip/connectedhomeip.git
     source-depth: 1
-    source-tag: v1.0.0.2
+    source-tag: v1.1.0.1
     source-submodules: []
     override-pull: |
       craftctl default
@@ -84,7 +84,20 @@ parts:
       source scripts/activate.sh
       set -u
       
-      cd examples/lighting-app/linux/
+      # Check SNAP_ARCH and install ZAP tool for arm builds
+      if [[ $SNAP_ARCH == "arm64" || $SNAP_ARCH == "armhf" ]]; then
+        ZAP_VERSION=v2023.05.04
+        set -x
+        mkdir -p /opt/zap-${ZAP_VERSION}
+        git clone https://github.com/project-chip/zap.git /opt/zap-${ZAP_VERSION}
+        cd /opt/zap-${ZAP_VERSION}
+        git checkout -b ${ZAP_VERSION}
+        npm ci
+        export ZAP_DEVELOPMENT_PATH=/opt/zap-${ZAP_VERSION}
+      fi
+
+      cd $CRAFT_PART_BUILD
+      cd ../../connectedhomeip/src/examples/lighting-app/linux/
 
       # Copy and replace the application files
       cp -vr $CRAFT_PART_BUILD/* .
@@ -96,6 +109,8 @@ parts:
       
       mkdir -p $CRAFT_PART_INSTALL/bin
       cp out/build/chip-lighting-app $CRAFT_PART_INSTALL/bin/lighting-app        
+    build-snaps:
+      - node/12/stable
     build-packages:
       - git
       - gcc

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -92,6 +92,9 @@ parts:
         git clone https://github.com/project-chip/zap.git /opt/zap-${ZAP_VERSION}
         cd /opt/zap-${ZAP_VERSION}
         git checkout -b ${ZAP_VERSION}
+        npm cache clean --force
+        npm install -g npm@9.7.1
+        npm update --force
         npm ci
         export ZAP_DEVELOPMENT_PATH=/opt/zap-${ZAP_VERSION}
       fi
@@ -110,7 +113,7 @@ parts:
       mkdir -p $CRAFT_PART_INSTALL/bin
       cp out/build/chip-lighting-app $CRAFT_PART_INSTALL/bin/lighting-app        
     build-snaps:
-      - node/12/stable
+      - node/20/stable
     build-packages:
       - git
       - gcc
@@ -129,6 +132,11 @@ parts:
       - libcairo2-dev
       - libreadline-dev
       - generate-ninja
+      - build-essential 
+      - libpango1.0-dev 
+      - libjpeg-dev 
+      - libgif-dev 
+      - librsvg2-dev
 
   local-bin:
     plugin: nil

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -12,7 +12,7 @@ confinement: strict
 base: core22
 architectures:
   - build-on: arm64
-  - build-on: armhf
+  # - build-on: armhf
   # - build-on: amd64
 
 layout:
@@ -79,28 +79,45 @@ parts:
       # Replace key-value store path:
       sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPPlatformConfig.h
 
-      # To avoid unrelated activation errors, don't treat unset variables as error
-      set +u
-      source scripts/activate.sh
-      set -u
-      
-      # Check SNAP_ARCH and install ZAP tool for arm builds
-      if [[ $SNAP_ARCH == "arm64" || $SNAP_ARCH == "armhf" ]]; then
-        ZAP_VERSION=v2023.05.04
+      # Install NodeJS and ZAP tool for arm builds
+      if [[ $SNAP_ARCH == "arm64" ]]; then
         set -x
+
+        mkdir node_js
+        cd node_js
+        wget https://nodejs.org/dist/v12.22.12/node-v12.22.12-linux-x64.tar.xz
+        tar xfvJ node-v12.22.12-linux-x64.tar.xz
+        cp -rn node-v12.22.12-linux-x64/. /opt/node-v12.22.12-linux-x64/
+        rm -r node-v12.22.12-linux-x64
+        rm -rf /opt/node
+        rm -rf /usr/bin/node
+        rm -rf /usr/bin/npm
+        rm -rf /usr/bin/npx
+        ln -s /opt/node-v12.22.12-linux-x64 /opt/node
+        ln -s /opt/node/bin/* /usr/bin
+        cd ..
+        rm -rf node_js
+
+        ZAP_VERSION=v2023.05.22-nightly
         mkdir -p /opt/zap-${ZAP_VERSION}
         git clone https://github.com/project-chip/zap.git /opt/zap-${ZAP_VERSION}
         cd /opt/zap-${ZAP_VERSION}
         git checkout -b ${ZAP_VERSION}
         npm cache clean --force
-        npm install -g npm@9.7.1
-        npm update --force
+        npm install -g npm@latest
         npm ci
         export ZAP_DEVELOPMENT_PATH=/opt/zap-${ZAP_VERSION}
       fi
 
       cd $CRAFT_PART_BUILD
-      cd ../../connectedhomeip/src/examples/lighting-app/linux/
+      cd ../../connectedhomeip/src
+
+      # To avoid unrelated activation errors, don't treat unset variables as error
+      set +u
+      source scripts/activate.sh
+      set -u
+
+      cd examples/lighting-app/linux/
 
       # Copy and replace the application files
       cp -vr $CRAFT_PART_BUILD/* .
@@ -112,8 +129,6 @@ parts:
       
       mkdir -p $CRAFT_PART_INSTALL/bin
       cp out/build/chip-lighting-app $CRAFT_PART_INSTALL/bin/lighting-app        
-    build-snaps:
-      - node/20/stable
     build-packages:
       - git
       - gcc
@@ -137,6 +152,7 @@ parts:
       - libjpeg-dev 
       - libgif-dev 
       - librsvg2-dev
+      - wget
 
   local-bin:
     plugin: nil

